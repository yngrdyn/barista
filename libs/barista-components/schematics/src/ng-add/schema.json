{
  "$schema": "http://json-schema.org/schema",
  "id": "barista-components-ng-add",
  "type": "object",
  "description": "Barista components ng-add schematic",
  "properties": {
    "project": {
      "type": "string",
      "description": "Name of the project where the Barista Components should be set up."
    },
    "module": {
      "type": "string",
      "subtype": "filepath",
      "description": "The path to the NgModule where the imports will be registered",
      "x-prompt": "The path to the NgModule where the imports will be registered:",
      "default": "src/app/app.module.ts",
      "$default": {
        "$source": "argv",
        "index": 0
      }
    },
    "animations": {
      "type": "boolean",
      "description": "Whether Angular browser animations should be set up.",
      "x-prompt": "Set up browser animations for Barista Components?",
      "default": true
    },
    "typography": {
      "type": "boolean",
      "description": "Whether to set up global typography styles.",
      "x-prompt": "Set up global Barista typography styles?",
      "default": false
    },
    "skipInstall": {
      "type": "boolean",
      "description": "Skip installing after adding @dynatrace/barista-components",
      "x-prompt": "Skip installing dependencies?",
      "default": false
    }
  },
  "required": [],
  "additionalProperties": false
}
