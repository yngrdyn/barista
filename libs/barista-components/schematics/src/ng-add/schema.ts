/**
 * @license
 * Copyright 2020 Dynatrace LLC
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export interface Schema {
  /** Name of the project where the Barista Components should be set up. */
  project: string;
  /** Whether Angular browser animations should be set up. */
  animations?: boolean;
  /** Whether to set up global typography styles. */
  typography?: boolean;
  /** The path to the NgModule where the imports will be registered */
  module?: string;
  /** Skip the install of the dependencies from the package.json */
  skipInstall?: boolean;
}

/** Schema with extended options that are needed for different rules */
export interface ExtendedSchema extends Schema {
  componentsVersion: string;
  peerDependencies: { [key: string]: string };
  migration: boolean;
}
