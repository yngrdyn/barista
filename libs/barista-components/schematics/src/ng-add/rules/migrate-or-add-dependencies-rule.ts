/**
 * @license
 * Copyright 2020 Dynatrace LLC
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import {
  chain,
  externalSchematic,
  Rule,
  SchematicContext,
} from '@angular-devkit/schematics';
import { Tree } from '@angular-devkit/schematics/src/tree/interface';
import { join } from 'path';
import { PackageJson } from '../../interfaces/package-json.interface';
import { updateJsonInTree } from '../../utils';
import { ExtendedSchema } from '../schema';

/** Check for angular/barista-components in package.json */
export function migrateOrAddDependenciesRule(options: ExtendedSchema): Rule {
  return (tree: Tree, context: SchematicContext) => {
    const rules: Rule[] = [
      // always add the new version of the barista components
      updateJsonInTree('package.json', (json: PackageJson) => {
        json.dependencies = json.dependencies || {};
        json.dependencies['@dynatrace/barista-components'] =
          options.componentsVersion;
        return json;
      }),
    ];

    for (const peerDependency of Object.keys(options.peerDependencies)) {
      const version = options.peerDependencies[peerDependency];
      rules.push(
        updateJsonInTree('package.json', (json: PackageJson) => {
          json.dependencies = json.dependencies || {};
          json.dependencies[peerDependency] = version;
          return json;
        }),
      );
    }

    // if a legacy version of the non open source version is installed
    // we have to migrate it to the @dynatrace/barista-components
    // installing the new package and refactoring all the imports
    if (options.migration) {
      // refactor all dt-iconpack and angular-components imports
      rules.push(migrateToVersion5(options));
      rules.push(
        updateJsonInTree('package.json', (json: PackageJson) => {
          json.dependencies = json.dependencies || {};
          delete json.dependencies['@dynatrace/angular-components'];
          return json;
        }),
      );
    }

    return chain(rules)(tree, context);
  };
}

/** Check filesystem for imports of dynatrace/angular-components and rename then to barista-components */
function migrateToVersion5(_options: ExtendedSchema): Rule {
  return (tree: Tree, context: SchematicContext) => {
    const rules: Rule[] = [];
    const collectionPath = join(__dirname, '../../../collection.json');
    // run external migration schematics
    rules.push(externalSchematic(collectionPath, 'update-5.0.0', {}));

    return chain(rules)(tree, context);
  };
}
