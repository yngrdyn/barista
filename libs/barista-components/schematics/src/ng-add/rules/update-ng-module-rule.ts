/**
 * @license
 * Copyright 2020 Dynatrace LLC
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Rule, SchematicContext, Tree } from '@angular-devkit/schematics';
import * as ts from 'typescript';
import {
  addImportToSourceFile,
  createPropertyAssignment,
  createStringLiteral,
  NgModuleProperties,
  printNode,
  readFileFromTree,
  updateNgModuleDecoratorProperties,
} from '../../utils';
import { ExtendedSchema } from '../schema';

/** The updateNgModuleRule modifies the app module */
export function updateNgModuleRule(options: ExtendedSchema): Rule {
  return (tree: Tree, _context: SchematicContext) => {
    // TODO: lukas.holzer check later on when no module is provided resolve it via the
    // workspace with the provided build target. project.architect.build.options.main
    // in the main file the module should be bootstrapped.

    if (!options.module || options.migration) {
      return;
    }

    const moduleSource = readFileFromTree(tree, options.module);

    const appModuleSourceFile = ts.createSourceFile(
      options.module,
      moduleSource,
      ts.ScriptTarget.Latest,
      true,
    );

    const iconModuleForRoot = ts.createCall(
      ts.createPropertyAccess(
        ts.createIdentifier('DtIconModule'),
        ts.createIdentifier('forRoot'),
      ),
      undefined,
      [
        ts.createObjectLiteral([
          createPropertyAssignment(
            'svgIconLocation',
            createStringLiteral('/assets/icons/{{name}}.svg', true),
          ),
        ]),
      ],
    );

    // Add the DtIconModule.forRoot and add the HttpClientModule
    const changes = [
      (sourceFile) =>
        updateNgModuleDecoratorProperties(
          sourceFile,
          NgModuleProperties.Imports,
          iconModuleForRoot,
        ),
      (sourceFile) =>
        updateNgModuleDecoratorProperties(
          sourceFile,
          NgModuleProperties.Imports,
          ts.createIdentifier('HttpClientModule'),
        ),
      (sourceFile) =>
        addImportToSourceFile(
          sourceFile,
          ['HttpClientModule'],
          '@angular/common/http',
        ),
      (sourceFile) =>
        addImportToSourceFile(
          sourceFile,
          ['DtIconModule'],
          '@dynatrace/barista-components/icon',
        ),
    ];

    // when the user doesn't want any animations add the NoopAnimationsModule
    const animationModule = options.animations
      ? 'BrowserAnimationsModule'
      : 'NoopAnimationsModule';

    // update of ngModule should be done before adding imports
    changes.unshift((sourceFile) =>
      updateNgModuleDecoratorProperties(
        sourceFile,
        NgModuleProperties.Imports,
        ts.createIdentifier(animationModule),
      ),
    );
    changes.push((sourceFile) =>
      addImportToSourceFile(
        sourceFile,
        [animationModule],
        '@angular/platform-browser/animations',
      ),
    );

    const modifiedSourceFile = modifySourceFile(appModuleSourceFile, changes);

    tree.overwrite(options.module, printNode(modifiedSourceFile));
  };
}

function modifySourceFile(
  sourceFile: ts.SourceFile,
  modifiers: Array<(source: ts.SourceFile) => ts.SourceFile>,
): ts.SourceFile {
  let prevSourceFile = sourceFile;

  for (const modifier of modifiers) {
    prevSourceFile = modifier(prevSourceFile);
  }

  return prevSourceFile;
}
