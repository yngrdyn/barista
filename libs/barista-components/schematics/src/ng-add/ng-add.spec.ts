/**
 * @license
 * Copyright 2020 Dynatrace LLC
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { noop, Tree, externalSchematic } from '@angular-devkit/schematics';
import { UnitTestTree } from '@angular-devkit/schematics/testing';
import {
  addFixtureToTree,
  addLegacyComponents,
  createWorkspace,
  runSchematic,
} from '../testing';
import { readFileFromTree, readJsonFromTree } from '../utils';
import { Schema } from './schema';

import { NodePackageInstallTask } from '@angular-devkit/schematics/tasks';
import { COULD_NOT_FIND_PROJECT_ERROR } from './rules';

export async function testNgAdd(
  testTree: Tree,
  options: Partial<Schema> = {},
): Promise<void> {
  const schemaOptions: Schema = {
    project: 'testProject',
    animations: true,
    module: '',
    typography: true,
    skipInstall: true,
    ...options,
  };
  await runSchematic('ng-add', schemaOptions, testTree);
}

let tree: UnitTestTree;

beforeEach(async () => {
  tree = await createWorkspace();
});

// Testing of Dynatrace Ng-Add Schematic
describe('Migrate existing angular-components to barista components', () => {
  let externalSchematicsSpy: jest.SpyInstance;

  beforeEach(async () => {
    externalSchematicsSpy = (externalSchematic as any) = jest
      .fn()
      .mockReturnValue(noop());
    await addLegacyComponents(tree);
  });

  afterEach(() => {
    externalSchematicsSpy.mockClear();
  });

  it('should call the migration schematic when legacy imports are detected', async () => {
    await testNgAdd(tree, { project: undefined });
    expect(externalSchematicsSpy).toBeCalledTimes(1);
    expect(externalSchematicsSpy).toBeCalledWith(
      expect.stringMatching(/collection\.json$/),
      'update-5.0.0',
      {},
    );
  });

  it('should update imports of @dynatrace/angular-components to barista-components in package.json', async () => {
    await testNgAdd(tree, { project: undefined });
    expect(readJsonFromTree(tree, '/package.json')).toMatchSnapshot();
  });

  it('should add barista icons to the angular.json', async () => {
    await addFixtureToTree(
      tree,
      'package-animations-existing.json',
      '/package.json',
    );

    await addFixtureToTree(tree, 'angular-simple.json', '/angular.json');
    await testNgAdd(tree, { project: 'myapp' });

    expect(readJsonFromTree(tree, '/angular.json')).toMatchSnapshot();
  });

  it('should include main.scss in angular.json, when typography is set to false ', async () => {
    await addFixtureToTree(
      tree,
      'package-animations-existing.json',
      '/package.json',
    );

    await addFixtureToTree(tree, 'angular-simple.json', '/angular.json');
    await testNgAdd(tree, { project: 'myapp', typography: false });

    expect(
      readJsonFromTree<any>(tree, '/angular.json').projects.myapp.architect
        .build.options.styles[1],
    ).toMatch(/main\.scss$/);
  });

  it('should add styles correctly even if there is no styles array in the angular.json ', async () => {
    await addFixtureToTree(
      tree,
      'package-animations-existing.json',
      '/package.json',
    );

    await addFixtureToTree(
      tree,
      'angular-without-styles.json',
      '/angular.json',
    );
    await testNgAdd(tree, { project: 'myapp' });

    expect(readJsonFromTree(tree, '/angular.json')).toMatchSnapshot();
  });

  it('should update the main ngModule with the correct providers', async () => {
    await testNgAdd(tree, {
      project: 'myapp',
      module: '/apps/myapp/src/app/app.module.ts',
    });

    expect(
      readFileFromTree(tree, '/apps/myapp/src/app/app.module.ts'),
    ).toMatchSnapshot();
  });

  it('should update the legacy angular json', async () => {
    await testNgAdd(tree, { project: 'myapp' });

    expect(readJsonFromTree(tree, '/angular.json')).toMatchSnapshot();
  });
});

describe('New workspace', () => {
  it('should add all the required modules in a fresh create angular app', async () => {
    await testNgAdd(tree, {
      project: 'myapp',
      module: '/apps/myapp/src/app/app.module.ts',
    });

    expect(
      readFileFromTree(tree, '/apps/myapp/src/app/app.module.ts'),
    ).toMatchSnapshot('app.module.ts');
    expect(readJsonFromTree(tree, '/angular.json')).toMatchSnapshot(
      'angular.json',
    );
    expect(readJsonFromTree(tree, '/package.json')).toMatchSnapshot(
      'package.json',
    );
  });

  it('should add import the No operation animations module from angular when animations is set to false', async () => {
    await addFixtureToTree(tree, 'package-empty.json', '/package.json');
    await testNgAdd(tree, {
      animations: false,
      project: 'myapp',
      module: '/apps/myapp/src/app/app.module.ts',
    });
    expect(
      readFileFromTree(tree, '/apps/myapp/src/app/app.module.ts'),
    ).toMatchSnapshot();
  });

  it('should add the `@angular/animations` package with the same version as the `@angular/core` package when specified', async () => {
    await addFixtureToTree(tree, 'package-empty.json', '/package.json');
    await testNgAdd(tree, { project: undefined });
    expect(readJsonFromTree(tree, '/package.json')).toMatchSnapshot();
  });

  it("shouldn't add @angular/animations` package if already installed", async () => {
    await addFixtureToTree(
      tree,
      'package-animations-existing.json',
      '/package.json',
    );

    await testNgAdd(tree, { animations: true, project: undefined });

    // check if the angular animations package is used more than once
    expect(
      readFileFromTree(tree, '/package.json').match(/\@angular\/animations/gim),
    ).toHaveLength(1);
    expect(readJsonFromTree(tree, '/package.json')).toMatchSnapshot();
  });

  it('should add @angular/platform-browser-dynamic', async () => {
    await addFixtureToTree(tree, 'package-empty.json', '/package.json');

    expect(readFileFromTree(tree, '/package.json')).not.toMatch(
      '@angular/platform-browser-dynamic',
    );

    await testNgAdd(tree, { project: undefined });
    expect(readJsonFromTree(tree, '/package.json')).toMatchSnapshot();
  });

  it('should install all the dependencies when skipInstall is set to false', async () => {
    const devkitTasksMock = ((NodePackageInstallTask as any) = jest
      .fn()
      .mockImplementation(() => ({
        toConfiguration: jest.fn().mockReturnValue({
          name: 'node-package',
        }),
      })));

    await testNgAdd(tree, { project: undefined, skipInstall: false });
    expect(devkitTasksMock).toHaveBeenCalledTimes(1);
  });

  it('should throw if a project is provided that does not exist', async () => {
    try {
      await testNgAdd(tree, { project: 'asdf' });
    } catch (e) {
      expect(e.message).toBe(COULD_NOT_FIND_PROJECT_ERROR('asdf'));
    } finally {
      expect.assertions(1);
    }
  });
});
