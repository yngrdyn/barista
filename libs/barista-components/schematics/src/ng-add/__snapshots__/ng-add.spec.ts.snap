// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Migrate existing angular-components to barista components should add barista icons to the angular.json 1`] = `
Object {
  "cli": Object {
    "defaultCollection": "@nrwl/angular",
  },
  "defaultProject": "myapp",
  "projects": Object {
    "myapp": Object {
      "architect": Object {
        "build": Object {
          "builder": "@angular-devkit/build-angular:browser",
          "options": Object {
            "aot": true,
            "assets": Array [
              "apps/myapp/src/favicon.ico",
              "apps/myapp/src/assets",
              Object {
                "glob": "**/*",
                "input": "node_modules/@dynatrace/barista-fonts/fonts/",
                "output": "/fonts",
              },
              Object {
                "glob": "metadata.json",
                "input": "node_modules/@dynatrace/barista-icons",
                "output": "/assets/icons",
              },
              Object {
                "glob": "*.svg",
                "input": "node_modules/@dynatrace/barista-icons",
                "output": "/assets/icons",
              },
            ],
            "index": "apps/myapp/src/index.html",
            "main": "apps/myapp/src/main.ts",
            "outputPath": "dist/apps/myapp",
            "polyfills": "apps/myapp/src/polyfills.ts",
            "scripts": Array [],
            "styles": Array [
              "node_modules/@dynatrace/barista-fonts/typography.scss",
              "node_modules/@dynatrace/barista-components/style/index.scss",
              "apps/myapp/src/styles.scss",
            ],
            "tsConfig": "apps/myapp/tsconfig.app.json",
          },
        },
      },
      "projectType": "application",
      "root": "apps/myapp",
      "sourceRoot": "apps/myapp/src",
    },
    "myapp-e2e": Object {
      "architect": Object {
        "e2e": Object {
          "builder": "@nrwl/cypress:cypress",
          "configurations": Object {
            "production": Object {
              "devServerTarget": "myapp:serve:production",
            },
          },
          "options": Object {
            "cypressConfig": "apps/myapp-e2e/cypress.json",
            "devServerTarget": "myapp:serve",
            "tsConfig": "apps/myapp-e2e/tsconfig.e2e.json",
          },
        },
        "lint": Object {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": Object {
            "exclude": Array [
              "**/node_modules/**",
              "!apps/myapp-e2e/**",
            ],
            "tsConfig": Array [
              "apps/myapp-e2e/tsconfig.e2e.json",
            ],
          },
        },
      },
      "projectType": "application",
      "root": "apps/myapp-e2e",
      "sourceRoot": "apps/myapp-e2e/src",
    },
    "some-lib": Object {
      "architect": Object {
        "lint": Object {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": Object {
            "exclude": Array [
              "**/node_modules/**",
              "!libs/some-lib/**",
            ],
            "tsConfig": Array [
              "libs/some-lib/tsconfig.lib.json",
              "libs/some-lib/tsconfig.spec.json",
            ],
          },
        },
        "test": Object {
          "builder": "@nrwl/jest:jest",
          "options": Object {
            "jestConfig": "libs/some-lib/jest.config.js",
            "setupFile": "libs/some-lib/src/test-setup.ts",
            "tsConfig": "libs/some-lib/tsconfig.spec.json",
          },
        },
      },
      "prefix": "mc",
      "projectType": "library",
      "root": "libs/some-lib",
      "schematics": Object {},
      "sourceRoot": "libs/some-lib/src",
    },
  },
  "version": 1,
}
`;

exports[`Migrate existing angular-components to barista components should add styles correctly even if there is no styles array in the angular.json  1`] = `
Object {
  "projects": Object {
    "myapp": Object {
      "architect": Object {
        "build": Object {
          "builder": "@angular-devkit/build-angular:browser",
          "options": Object {
            "assets": Array [
              Object {
                "glob": "**/*",
                "input": "node_modules/@dynatrace/barista-fonts/fonts/",
                "output": "/fonts",
              },
            ],
            "index": "apps/myapp/src/index.html",
            "main": "apps/myapp/src/main.ts",
            "outputPath": "dist/apps/myapp",
            "polyfills": "apps/myapp/src/polyfills.ts",
            "styles": Array [
              "node_modules/@dynatrace/barista-fonts/typography.scss",
              "node_modules/@dynatrace/barista-components/style/index.scss",
            ],
            "tsConfig": "apps/myapp/tsconfig.app.json",
          },
        },
      },
      "projectType": "application",
      "root": "apps/myapp",
      "sourceRoot": "apps/myapp/src",
    },
  },
  "version": 1,
}
`;

exports[`Migrate existing angular-components to barista components should update imports of @dynatrace/angular-components to barista-components in package.json 1`] = `
Object {
  "dependencies": Object {
    "@angular/animations": "^8.2.12",
    "@angular/cdk": "^8.2.12",
    "@angular/common": "^8.2.12",
    "@angular/core": "^8.2.12",
    "@dynatrace/barista-components": "^5.0.0",
    "@dynatrace/barista-fonts": ">= 1.0.0 < 2",
    "@dynatrace/barista-icons": ">= 3.0.0 < 4",
    "@types/highcharts": "^5.0.23",
    "d3-scale": "^3.0.0",
    "d3-shape": "^1.3.5",
    "highcharts": "^6.0.7",
  },
}
`;

exports[`Migrate existing angular-components to barista components should update the legacy angular json 1`] = `
Object {
  "projects": Object {
    "myapp": Object {
      "architect": Object {
        "build": Object {
          "builder": "@angular-devkit/build-angular:browser",
          "options": Object {
            "aot": true,
            "assets": Array [
              "apps/myapp/src/favicon.ico",
              "apps/myapp/src/assets",
              Object {
                "glob": "**/*",
                "input": "node_modules/@dynatrace/barista-fonts/fonts/",
                "output": "/fonts",
              },
              Object {
                "glob": "**",
                "input": "node_modules/@dynatrace/barista-components/assets",
                "output": "/cache",
              },
              Object {
                "glob": "*.svg",
                "input": "node_modules/@dynatrace/barista-icons",
                "output": "/cache/icons",
              },
            ],
            "index": "apps/myapp/src/index.html",
            "main": "apps/myapp/src/main.ts",
            "outputPath": "dist/apps/myapp",
            "polyfills": "apps/myapp/src/polyfills.ts",
            "scripts": Array [],
            "styles": Array [
              "node_modules/@dynatrace/barista-fonts/typography.scss",
              "apps/myapp/src/styles.scss",
            ],
            "tsConfig": "apps/myapp/tsconfig.app.json",
          },
        },
      },
      "projectType": "application",
      "root": "apps/myapp",
      "sourceRoot": "apps/myapp/src",
    },
  },
  "version": 1,
}
`;

exports[`Migrate existing angular-components to barista components should update the main ngModule with the correct providers 1`] = `
"import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';

import { AppComponent } from './app.component';

@NgModule({
  declarations: [
    AppComponent
  ],
  imports: [
    BrowserModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
"
`;

exports[`New workspace should add @angular/platform-browser-dynamic 1`] = `
Object {
  "dependencies": Object {
    "@angular/animations": "^8.2.12",
    "@angular/cdk": "^8.2.12",
    "@angular/common": "^8.2.12",
    "@angular/compiler": "^8.2.12",
    "@angular/core": "^8.2.12",
    "@dynatrace/barista-components": "^5.0.0",
    "@dynatrace/barista-fonts": ">= 1.0.0 < 2",
    "@dynatrace/barista-icons": ">= 3.0.0 < 4",
    "@types/highcharts": "^5.0.23",
    "d3-scale": "^3.0.0",
    "d3-shape": "^1.3.5",
    "highcharts": "^6.0.7",
  },
  "devDependencies": Object {
    "@angular-devkit/build-angular": "^0.803.15",
    "@angular-devkit/build-webpack": "^0.803.15",
    "@angular-devkit/schematics": "^8.3.15",
    "@angular/cli": "^8.3.15",
  },
}
`;

exports[`New workspace should add all the required modules in a fresh create angular app: angular.json 1`] = `
Object {
  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
  "defaultProject": "myapp",
  "newProjectRoot": "apps",
  "projects": Object {
    "myapp": Object {
      "architect": Object {
        "build": Object {
          "builder": "@angular-devkit/build-angular:browser",
          "configurations": Object {
            "production": Object {
              "budgets": Array [
                Object {
                  "maximumError": "5mb",
                  "maximumWarning": "2mb",
                  "type": "initial",
                },
                Object {
                  "maximumError": "10kb",
                  "maximumWarning": "6kb",
                  "type": "anyComponentStyle",
                },
              ],
              "buildOptimizer": true,
              "extractCss": true,
              "extractLicenses": true,
              "fileReplacements": Array [
                Object {
                  "replace": "apps/myapp/src/environments/environment.ts",
                  "with": "apps/myapp/src/environments/environment.prod.ts",
                },
              ],
              "namedChunks": false,
              "optimization": true,
              "outputHashing": "all",
              "sourceMap": false,
              "vendorChunk": false,
            },
          },
          "options": Object {
            "aot": true,
            "assets": Array [
              "apps/myapp/src/favicon.ico",
              "apps/myapp/src/assets",
              Object {
                "glob": "**/*",
                "input": "node_modules/@dynatrace/barista-fonts/fonts/",
                "output": "/fonts",
              },
              Object {
                "glob": "metadata.json",
                "input": "node_modules/@dynatrace/barista-icons",
                "output": "/assets/icons",
              },
              Object {
                "glob": "*.svg",
                "input": "node_modules/@dynatrace/barista-icons",
                "output": "/assets/icons",
              },
            ],
            "index": "apps/myapp/src/index.html",
            "main": "apps/myapp/src/main.ts",
            "outputPath": "dist/myapp",
            "polyfills": "apps/myapp/src/polyfills.ts",
            "scripts": Array [],
            "styles": Array [
              "node_modules/@dynatrace/barista-fonts/typography.scss",
              "node_modules/@dynatrace/barista-components/style/index.scss",
              "apps/myapp/src/styles.scss",
            ],
            "tsConfig": "apps/myapp/tsconfig.app.json",
          },
        },
        "e2e": Object {
          "builder": "@angular-devkit/build-angular:protractor",
          "configurations": Object {
            "production": Object {
              "devServerTarget": "myapp:serve:production",
            },
          },
          "options": Object {
            "devServerTarget": "myapp:serve",
            "protractorConfig": "apps/myapp/e2e/protractor.conf.js",
          },
        },
        "extract-i18n": Object {
          "builder": "@angular-devkit/build-angular:extract-i18n",
          "options": Object {
            "browserTarget": "myapp:build",
          },
        },
        "lint": Object {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": Object {
            "exclude": Array [
              "**/node_modules/**",
            ],
            "tsConfig": Array [
              "apps/myapp/tsconfig.app.json",
              "apps/myapp/tsconfig.spec.json",
              "apps/myapp/e2e/tsconfig.json",
            ],
          },
        },
        "serve": Object {
          "builder": "@angular-devkit/build-angular:dev-server",
          "configurations": Object {
            "production": Object {
              "browserTarget": "myapp:build:production",
            },
          },
          "options": Object {
            "browserTarget": "myapp:build",
          },
        },
        "test": Object {
          "builder": "@angular-devkit/build-angular:karma",
          "options": Object {
            "assets": Array [
              "apps/myapp/src/favicon.ico",
              "apps/myapp/src/assets",
            ],
            "karmaConfig": "apps/myapp/karma.conf.js",
            "main": "apps/myapp/src/test.ts",
            "polyfills": "apps/myapp/src/polyfills.ts",
            "scripts": Array [],
            "styles": Array [
              "apps/myapp/src/styles.scss",
            ],
            "tsConfig": "apps/myapp/tsconfig.spec.json",
          },
        },
      },
      "prefix": "app",
      "projectType": "application",
      "root": "apps/myapp",
      "schematics": Object {
        "@schematics/angular:class": Object {
          "skipTests": true,
        },
        "@schematics/angular:component": Object {
          "skipTests": true,
          "style": "scss",
        },
        "@schematics/angular:directive": Object {
          "skipTests": true,
        },
        "@schematics/angular:guard": Object {
          "skipTests": true,
        },
        "@schematics/angular:interceptor": Object {
          "skipTests": true,
        },
        "@schematics/angular:module": Object {
          "skipTests": true,
        },
        "@schematics/angular:pipe": Object {
          "skipTests": true,
        },
        "@schematics/angular:service": Object {
          "skipTests": true,
        },
      },
      "sourceRoot": "apps/myapp/src",
    },
  },
  "version": 1,
}
`;

exports[`New workspace should add all the required modules in a fresh create angular app: app.module.ts 1`] = `
"import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';
import { AppComponent } from './app.component';
import { HttpClientModule } from '@angular/common/http';
import { DtIconModule } from '@dynatrace/barista-components/icon';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
@NgModule({
    declarations: [
        AppComponent
    ],
    imports: [
        BrowserModule,
        BrowserAnimationsModule,
        DtIconModule.forRoot({ svgIconLocation: '/assets/icons/{{name}}.svg' }),
        HttpClientModule
    ],
    providers: [],
    bootstrap: [AppComponent]
})
export class AppModule {
}
"
`;

exports[`New workspace should add all the required modules in a fresh create angular app: package.json 1`] = `
Object {
  "dependencies": Object {
    "@angular/animations": "^8.2.12",
    "@angular/cdk": "^8.2.12",
    "@angular/common": "^8.2.12",
    "@angular/compiler": "~10.0.5",
    "@angular/core": "^8.2.12",
    "@angular/forms": "~10.0.5",
    "@angular/platform-browser": "~10.0.5",
    "@angular/platform-browser-dynamic": "~10.0.5",
    "@angular/router": "~10.0.5",
    "@dynatrace/barista-components": "^5.0.0",
    "@dynatrace/barista-fonts": ">= 1.0.0 < 2",
    "@dynatrace/barista-icons": ">= 3.0.0 < 4",
    "@types/highcharts": "^5.0.23",
    "d3-scale": "^3.0.0",
    "d3-shape": "^1.3.5",
    "highcharts": "^6.0.7",
    "rxjs": "~6.5.5",
    "tslib": "^2.0.0",
    "zone.js": "~0.10.3",
  },
  "devDependencies": Object {
    "@angular-devkit/build-angular": "~0.1000.4",
    "@angular/cli": "~8.0.0",
    "@angular/compiler-cli": "~10.0.5",
    "@types/jasmine": "~3.5.0",
    "@types/jasminewd2": "~2.0.3",
    "@types/node": "^12.11.1",
    "codelyzer": "^6.0.0",
    "jasmine-core": "~3.5.0",
    "jasmine-spec-reporter": "~5.0.0",
    "karma": "~5.0.0",
    "karma-chrome-launcher": "~3.1.0",
    "karma-coverage-istanbul-reporter": "~3.0.2",
    "karma-jasmine": "~3.3.0",
    "karma-jasmine-html-reporter": "^1.5.0",
    "protractor": "~7.0.0",
    "ts-node": "~8.3.0",
    "tslint": "~6.1.0",
    "typescript": "~3.9.5",
  },
  "name": "workspace",
  "private": true,
  "scripts": Object {
    "build": "ng build",
    "e2e": "ng e2e",
    "lint": "ng lint",
    "ng": "ng",
    "start": "ng serve",
    "test": "ng test",
  },
  "version": "0.0.0",
}
`;

exports[`New workspace should add import the No operation animations module from angular when animations is set to false 1`] = `
"import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';
import { AppComponent } from './app.component';
import { HttpClientModule } from '@angular/common/http';
import { DtIconModule } from '@dynatrace/barista-components/icon';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
@NgModule({
    declarations: [
        AppComponent
    ],
    imports: [
        BrowserModule,
        NoopAnimationsModule,
        DtIconModule.forRoot({ svgIconLocation: '/assets/icons/{{name}}.svg' }),
        HttpClientModule
    ],
    providers: [],
    bootstrap: [AppComponent]
})
export class AppModule {
}
"
`;

exports[`New workspace should add the \`@angular/animations\` package with the same version as the \`@angular/core\` package when specified 1`] = `
Object {
  "dependencies": Object {
    "@angular/animations": "^8.2.12",
    "@angular/cdk": "^8.2.12",
    "@angular/common": "^8.2.12",
    "@angular/compiler": "^8.2.12",
    "@angular/core": "^8.2.12",
    "@dynatrace/barista-components": "^5.0.0",
    "@dynatrace/barista-fonts": ">= 1.0.0 < 2",
    "@dynatrace/barista-icons": ">= 3.0.0 < 4",
    "@types/highcharts": "^5.0.23",
    "d3-scale": "^3.0.0",
    "d3-shape": "^1.3.5",
    "highcharts": "^6.0.7",
  },
  "devDependencies": Object {
    "@angular-devkit/build-angular": "^0.803.15",
    "@angular-devkit/build-webpack": "^0.803.15",
    "@angular-devkit/schematics": "^8.3.15",
    "@angular/cli": "^8.3.15",
  },
}
`;

exports[`New workspace shouldn't add @angular/animations\` package if already installed 1`] = `
Object {
  "dependencies": Object {
    "@angular/animations": "^8.2.12",
    "@angular/cdk": "^8.2.12",
    "@angular/common": "^8.2.12",
    "@angular/core": "^8.2.12",
    "@dynatrace/barista-components": "^5.0.0",
    "@dynatrace/barista-fonts": ">= 1.0.0 < 2",
    "@dynatrace/barista-icons": ">= 3.0.0 < 4",
    "@types/highcharts": "^5.0.23",
    "d3-scale": "^3.0.0",
    "d3-shape": "^1.3.5",
    "highcharts": "^6.0.7",
  },
}
`;
